src/main/java/org/candlepin/subscriptions/tally/admin/InternalTallyResource.java

```diff
@@ -111,4 +124,157 @@ public void purgeTallySnapshots() {
       log.warn("A tally snapshots purge job is already running.");
     }
   }
+
+  /**
+   * Clear tallies, hosts, and events for a given org ID. Enabled via ENABLE_ACCOUNT_RESET
+   * environment variable. Intended only for non-prod environments.
+   *
+   * @param orgId Red Hat orgId
+   */
+  @Override
+  public String deleteDataAssociatedWithOrg(String orgId) {
+    if (isFeatureEnabled()) {
+      log.info("Received request to delete all data associated with orgId {}", orgId);
+      try {
+        internalTallyDataController.deleteDataAssociatedWithOrg(orgId);
+      } catch (Exception e) {
+        log.error("Unable to delete data for organization {} due to {}", orgId, e);
+        return String.format("Unable to delete data for organization %s", orgId);
+      }
+      var successMessage = "Finished deleting data associated with organization " + orgId;
+      log.info(successMessage);
+      return successMessage;
+    } else {
+      return FEATURE_NOT_ENABLED_MESSSAGE;
+    }
+  }
+
+  /**
+   * Delete an event. Supported only in dev-mode.
+   *
+   * @param eventId Event Id
+   * @return success or error message
+   */
+  @Override
+  public String deleteEvent(String eventId) {
+    if (isFeatureEnabled()) {
+      try {
+        internalTallyDataController.deleteEvent(eventId);
+        return String.format("Successfully deleted Event with ID: %s", eventId);
+      } catch (Exception e) {
+        log.error("Failed to delete Event with ID: {}  Cause: {}", eventId, e.getMessage());
+        return String.format("Failed to delete Event with ID: %s ", eventId);
+      }
+    } else {
+      return FEATURE_NOT_ENABLED_MESSSAGE;
+    }
+  }
+
+  /**
+   * Fetch events by org
+   *
+   * @param orgId Red Hat orgId
+   * @param begin Beginning of time range (inclusive)
+   * @param end End of time range (exclusive)
+   * @return success or error message
+   */
+  @Override
+  public String fetchEventsForOrgIdInTimeRange(
+      String orgId, OffsetDateTime begin, OffsetDateTime end) {
+    try {
+      return internalTallyDataController.fetchEventsForOrgIdInTimeRange(orgId, begin, end);
+    } catch (Exception e) {
+      log.error("Unable to deserialize event list ", e);
+      return "Unable to deserialize event list";
+    }
+  }
+
+  /**
+   * Save a list of events. Supported only in dev-mode.
+   *
+   * @param jsonListOfEvents Event list specified as JSON
+   * @return success or error message
+   */
+  @Override
+  public String saveEvents(String jsonListOfEvents) {
+    if (isFeatureEnabled()) {
+      return internalTallyDataController.saveEvents(jsonListOfEvents);
+    } else {
+      return FEATURE_NOT_ENABLED_MESSSAGE;
+    }
+  }
+
+  /** Update tally snapshots for all orgs */
+  @Override
+  public void tallyConfiguredOrgs() {
```

Comment:
Same as with the strings, instead we return an object called DefaultResponse with a success status

Reply:
Done